src/Applications/LDAS_App/util/inputs/obs_scaling_params/get_model_and_obs_clim_stats_latlon_grid.m

```diff
@@ -0,0 +1,351 @@
+                    '_W_', num2str(round(w_days/5)), 'p_Nmin_', num2str(Ndata_min)];
+
+%======================================================
+
+% Define lat/lon grid (dateline-on-edge, pole-on-edge)
+% Define lower-left corner coordinates and grid cell size
+ll_lon = -180;
+ll_lat =  -90;
+
+d_lon = resol;
+d_lat = resol;
+
+% Calculate number of longitude and latitude grid cells
+n_lon = round(360 / d_lon);
+n_lat = round(180 / d_lat);
+
+% Calculate longitude and latitude values for the grid
+ll_lons = linspace(ll_lon, ll_lon + (n_lon-1)*d_lon, n_lon);
+ll_lats = linspace(ll_lat, ll_lat + (n_lat-1)*d_lat, n_lat);
+
+% Create grid index
+obsnum         = (1:n_lon*n_lat)';
+[i_out, j_out] = ind2sub([n_lon, n_lat], obsnum);
+lon_out        = ll_lons(i_out)';
+lat_out        = ll_lats(j_out)';
+N_gridcells    = length(obsnum);
+  
+% initialize output statistics
+o_data_sum  = NaN(N_species, N_gridcells, w_days);
+m_data_sum  = NaN(N_species, N_gridcells, w_days);
+o_data_sum2 = NaN(N_species, N_gridcells, w_days);
+m_data_sum2 = NaN(N_species, N_gridcells, w_days);
+m_data_min  = NaN(N_species, N_gridcells, w_days);
+m_data_max  = NaN(N_species, N_gridcells, w_days);
+N_data      = NaN(N_species, N_gridcells, w_days);
+
+data_out    = NaN(N_species, Nf, N_gridcells, N_pentads);
+data2D      = NaN(Nf, N_gridcells);
+
+% -------------------------------------------------------------		  
+
+% make sure t0_assim is *first* analysis time in a day
+
+t0_assim = mod( t0_assim, dt_assim );
+
+count = 0;
+
+for imonth = 1:length(run_months)
+
+  month = run_months(imonth);
+  
+  for day = 1:days_in_month( 2014, month) %2014 = random non-leap year
+    
+    if count < w_days
+      count = count + 1;
+    else
+      count = w_days;
+    end
+    
+    for seconds_in_day = t0_assim:dt_assim:(86400-1)
+      
+      hour    = floor( seconds_in_day/3600);
+      minute  = floor((seconds_in_day-hour*3600)/60);
+      seconds = seconds_in_day-hour*3600-minute*60;
+      
+      if (seconds ~= 0)
+        input('something is wrong! Ctrl-c now')
+      end
+      
+      for year = start_year(imonth):end_year(imonth)
+        
+        YYYYMMDD = [num2str(year, '%4.4d'), num2str(month, '%2.2d'), num2str(day, '%2.2d')];
+        HHMM = [num2str(hour, '%2.2d'), num2str(minute, '%2.2d')];
+        
+        % read innov files
+        fname = [inpath, '/ana/ens_avg/', 'Y', YYYYMMDD(1:4), '/', 'M', YYYYMMDD(5:6), '/', exp_run, '.ens_avg.ldas_ObsFcstAna.', YYYYMMDD, '_', HHMM, 'z.bin'];
+        ifp = fopen(fname, 'r', 'l');
+        
+        if (ifp > 0) % Proceed only if file exists
+          fclose(ifp);
+          [date_time, obs_assim, obs_species, obs_tilenum, obs_lon, obs_lat, obs_obs, obs_obsvar, obs_fcst, obs_fcstvar, obs_ana, obs_anavar] = read_ObsFcstAna(fname);
+          
+          % remove tiles when there is no obs_fcst (obs_fcst == 0 in innov output when missing)
+          idx = find(obs_fcst == 0);
```

Comment:
@gmao-qliu @amfox37: 
Not sure why the check here is for "obs_fcst==0".  Do we not have a defined no-data-value for obs_fcst?  The zero value might work for ASCAT sfds, but it's easy to imagine that zero is problematic if in future we're trying to generalize the subroutine for other variables (e.g., SWE).  
My guess is that obs_fcst is -9999 (SMAP no-data-value) or possibly 1.e15 (MAPL_UNDEF).
Having said that, we probably don't even need to keep this block.  The matlab reader for "ObsFcstAna" files produces NaN for no-data.  I think this is inherited from "get_model_and_obs_clim_stats.m" and would need to be changed there as well.

Reply:
@amfox37, @gmao-qliu : With any luck, my commit https://github.com/GEOS-ESM/GEOSldas/pull/656/commits/565c8682e3c1aead0dfc7263e89eee4110336138 fixes this in both `get_model_and_obs_clim_stats*.m` scripts.  Please let me know if you see any problems.